package/kernel/linux/modules/crypto.mk

```diff
@@ -446,17 +446,30 @@ endef
 
 $(eval $(call KernelPackage,crypto-kpp))
 
-
-define KernelPackage/crypto-lib-blake2s
+define KernelPackage/crypto-lib-blake2s/default
   TITLE:=BLAKE2s hash function library
   KCONFIG:=CONFIG_CRYPTO_LIB_BLAKE2S
   HIDDEN:=1
-  FILES:= \
-	$(LINUX_DIR)/lib/crypto/libblake2s.ko \
-	$(LINUX_DIR)/lib/crypto/libblake2s-generic.ko
+  FILES:=$(LINUX_DIR)/lib/crypto/libblake2s.ko
   $(call AddDepends/crypto,+PACKAGE_kmod-crypto-hash:kmod-crypto-hash)
 endef
 
+define KernelPackage/crypto-lib-blake2s/generic
+  KCONFIG+=CONFIG_CRYPTO_LIB_BLAKE2S_GENERIC@ge5.15
+  FILES+=$(LINUX_DIR)/lib/crypto/libblake2s-generic.ko
+endef
+
+ifeq ($(ARCH),$(filter $(ARCH),arm x86_64))
```

Comment:
@pkgadd talking about this... your x86 is 64bit correct? not 32bit

can you try add i386 to the list?

Reply:
Will do, testing the following diffconfig now (build-test only, not using any i386 systems with OpenWrt myself, so this is a mindless x86_64 --> x86_generic config conversion):
```
CONFIG_TARGET_x86=y
CONFIG_TARGET_x86_generic=y
CONFIG_TARGET_MULTI_PROFILE=y
CONFIG_TARGET_DEVICE_x86_generic_DEVICE_generic=y
CONFIG_TARGET_DEVICE_PACKAGES_x86_generic_DEVICE_generic="slh-misc-32m slh-misc-luci-statistics slh-misc-qos irqbalance -libustream-wolfssl -ca-bundle -wpad-basic -wpad-mini -wpad wpad-openssl -wpad-basic-wolfssl"
CONFIG_DEVEL=y
CONFIG_TARGET_PER_DEVICE_ROOTFS=y
CONFIG_ATH9K_SUPPORT_PCOEM=y
CONFIG_ATH_USER_REGD=y
CONFIG_BUILD_LOG=y
CONFIG_DRIVER_11AC_SUPPORT=y
CONFIG_DRIVER_11N_SUPPORT=y
CONFIG_DROPBEAR_ECC=y
CONFIG_DROPBEAR_ECC_FULL=y
CONFIG_HTOP_LMSENSORS=y
CONFIG_IMAGEOPT=y
CONFIG_INCLUDE_CONFIG=y
# CONFIG_KERNEL_DEBUG_INFO is not set
# CONFIG_KERNEL_DEBUG_KERNEL is not set
CONFIG_LUCI_LANG_de=m
CONFIG_LUCI_LANG_en=m
CONFIG_OPENSSL_ENGINE=y
CONFIG_OPENSSL_NO_DEPRECATED=y
CONFIG_OPENSSL_OPTIMIZE_SPEED=y
CONFIG_OPENSSL_PREFER_CHACHA_OVER_GCM=y
CONFIG_OPENSSL_WITH_ASM=y
CONFIG_OPENSSL_WITH_CHACHA_POLY1305=y
CONFIG_OPENSSL_WITH_CMS=y
CONFIG_OPENSSL_WITH_ERROR_MESSAGES=y
CONFIG_OPENSSL_WITH_PSK=y
CONFIG_OPENSSL_WITH_SRP=y
CONFIG_OPENSSL_WITH_SSE2=y
CONFIG_OPENSSL_WITH_TLS13=y
CONFIG_PACKAGE_6in4=m
CONFIG_PACKAGE_ATH_DFS=y
CONFIG_PACKAGE_MAC80211_DEBUGFS=y
CONFIG_PACKAGE_MAC80211_MESH=y
CONFIG_PACKAGE_NTFS-3G_HAS_PROBE=y
CONFIG_PACKAGE_adblock=m
CONFIG_PACKAGE_ath9k-htc-firmware=y
CONFIG_PACKAGE_block-mount=m
CONFIG_PACKAGE_blockd=m
CONFIG_PACKAGE_ca-bundle=m
CONFIG_PACKAGE_ca-certificates=m
CONFIG_PACKAGE_cgi-io=m
CONFIG_PACKAGE_collectd=m
CONFIG_PACKAGE_collectd-mod-conntrack=m
CONFIG_PACKAGE_collectd-mod-cpu=m
CONFIG_PACKAGE_collectd-mod-cpufreq=m
CONFIG_PACKAGE_collectd-mod-entropy=m
CONFIG_PACKAGE_collectd-mod-exec=m
CONFIG_PACKAGE_collectd-mod-interface=m
CONFIG_PACKAGE_collectd-mod-iwinfo=m
CONFIG_PACKAGE_collectd-mod-load=m
CONFIG_PACKAGE_collectd-mod-memory=m
CONFIG_PACKAGE_collectd-mod-network=m
CONFIG_PACKAGE_collectd-mod-ping=m
CONFIG_PACKAGE_collectd-mod-rrdtool=m
CONFIG_PACKAGE_collectd-mod-snmp=m
CONFIG_PACKAGE_collectd-mod-sqm=m
CONFIG_PACKAGE_collectd-mod-thermal=m
CONFIG_PACKAGE_collectd-mod-uptime=m
CONFIG_PACKAGE_collectd-mod-wireless=m
CONFIG_PACKAGE_coreutils=m
CONFIG_PACKAGE_coreutils-sort=m
CONFIG_PACKAGE_ddns-scripts=m
CONFIG_PACKAGE_ddns-scripts-noip=m
CONFIG_PACKAGE_ddns-scripts-services=m
CONFIG_PACKAGE_diffutils=m
CONFIG_PACKAGE_eject=m
CONFIG_PACKAGE_etherwake=m
CONFIG_PACKAGE_ethtool-full=m
CONFIG_PACKAGE_exfat-fsck=m
CONFIG_PACKAGE_exfat-mkfs=m
CONFIG_PACKAGE_fdisk=y
CONFIG_PACKAGE_gdisk=y
CONFIG_PACKAGE_hostapd-common=y
CONFIG_PACKAGE_hostapd-utils=m
CONFIG_PACKAGE_htop=m
CONFIG_PACKAGE_intel-microcode=y
CONFIG_PACKAGE_ip-bridge=m
CONFIG_PACKAGE_ip-full=y
CONFIG_PACKAGE_iperf3=y
CONFIG_PACKAGE_iptables-mod-ipopt=m
CONFIG_PACKAGE_iptables-nft=m
CONFIG_PACKAGE_irqbalance=m
CONFIG_PACKAGE_iw=y
CONFIG_PACKAGE_iwinfo=y
CONFIG_PACKAGE_iwlwifi-firmware-ax200=y
CONFIG_PACKAGE_iwlwifi-firmware-ax210=y
CONFIG_PACKAGE_kmod-alx=y
CONFIG_PACKAGE_kmod-ath=y
CONFIG_PACKAGE_kmod-ath5k=y
CONFIG_PACKAGE_kmod-ath9k=y
CONFIG_PACKAGE_kmod-ath9k-common=y
CONFIG_PACKAGE_kmod-ath9k-htc=y
CONFIG_PACKAGE_kmod-cfg80211=y
CONFIG_PACKAGE_kmod-crypto-aead=y
CONFIG_PACKAGE_kmod-crypto-ccm=y
CONFIG_PACKAGE_kmod-crypto-cmac=y
CONFIG_PACKAGE_kmod-crypto-ctr=y
CONFIG_PACKAGE_kmod-crypto-gcm=y
CONFIG_PACKAGE_kmod-crypto-gf128=y
CONFIG_PACKAGE_kmod-crypto-ghash=y
CONFIG_PACKAGE_kmod-crypto-hmac=y
CONFIG_PACKAGE_kmod-crypto-kpp=m
CONFIG_PACKAGE_kmod-crypto-lib-blake2s=m
CONFIG_PACKAGE_kmod-crypto-lib-chacha20=m
CONFIG_PACKAGE_kmod-crypto-lib-chacha20poly1305=m
CONFIG_PACKAGE_kmod-crypto-lib-curve25519=m
CONFIG_PACKAGE_kmod-crypto-lib-poly1305=m
CONFIG_PACKAGE_kmod-crypto-manager=y
CONFIG_PACKAGE_kmod-crypto-null=y
CONFIG_PACKAGE_kmod-crypto-rng=y
CONFIG_PACKAGE_kmod-crypto-seqiv=y
CONFIG_PACKAGE_kmod-crypto-sha256=y
CONFIG_PACKAGE_kmod-eeprom-93cx6=y
CONFIG_PACKAGE_kmod-fs-autofs4=m
CONFIG_PACKAGE_kmod-fs-exfat=m
CONFIG_PACKAGE_kmod-fs-ext4=m
CONFIG_PACKAGE_kmod-fuse=m
CONFIG_PACKAGE_kmod-ifb=m
CONFIG_PACKAGE_kmod-ikconfig=m
CONFIG_PACKAGE_kmod-ipt-core=m
CONFIG_PACKAGE_kmod-ipt-ipopt=m
CONFIG_PACKAGE_kmod-iptunnel=m
CONFIG_PACKAGE_kmod-iptunnel4=m
CONFIG_PACKAGE_kmod-iwlwifi=y
CONFIG_PACKAGE_kmod-lib-crc16=m
CONFIG_PACKAGE_kmod-mac80211=y
CONFIG_PACKAGE_kmod-mdio=y
CONFIG_PACKAGE_kmod-nf-conntrack-netlink=m
CONFIG_PACKAGE_kmod-nf-ipt=m
CONFIG_PACKAGE_kmod-nft-compat=m
CONFIG_PACKAGE_kmod-rt2800-lib=y
CONFIG_PACKAGE_kmod-rt2800-mmio=y
CONFIG_PACKAGE_kmod-rt2800-pci=y
CONFIG_PACKAGE_kmod-rt2800-usb=y
CONFIG_PACKAGE_kmod-rt2x00-lib=y
CONFIG_PACKAGE_kmod-rt2x00-mmio=y
CONFIG_PACKAGE_kmod-rt2x00-pci=y
CONFIG_PACKAGE_kmod-rt2x00-usb=y
CONFIG_PACKAGE_kmod-sched-cake=m
CONFIG_PACKAGE_kmod-sched-core=m
CONFIG_PACKAGE_kmod-scsi-core=m
CONFIG_PACKAGE_kmod-sit=m
CONFIG_PACKAGE_kmod-udptunnel4=m
CONFIG_PACKAGE_kmod-udptunnel6=m
CONFIG_PACKAGE_kmod-usb-core=y
CONFIG_PACKAGE_kmod-usb-ehci=m
CONFIG_PACKAGE_kmod-usb-printer=m
CONFIG_PACKAGE_kmod-usb-storage=m
CONFIG_PACKAGE_kmod-usb-storage-uas=m
CONFIG_PACKAGE_kmod-usb2=m
CONFIG_PACKAGE_kmod-wireguard=m
CONFIG_PACKAGE_libbpf=y
CONFIG_PACKAGE_libelf=y
CONFIG_PACKAGE_libevdev=y
CONFIG_PACKAGE_libfdisk=y
CONFIG_PACKAGE_libgcrypt=m
CONFIG_PACKAGE_libgpg-error=m
CONFIG_PACKAGE_libiptext=m
CONFIG_PACKAGE_libiptext-nft=m
CONFIG_PACKAGE_libiptext6=m
CONFIG_PACKAGE_libiwinfo=y
CONFIG_PACKAGE_libiwinfo-data=y
CONFIG_PACKAGE_libiwinfo-lua=m
CONFIG_PACKAGE_libkmod=y
CONFIG_PACKAGE_libltdl=m
CONFIG_PACKAGE_liblua=m
CONFIG_PACKAGE_liblucihttp=m
CONFIG_PACKAGE_liblucihttp-lua=m
CONFIG_PACKAGE_libmount=y
CONFIG_PACKAGE_libncurses=y
CONFIG_PACKAGE_libnetsnmp=m
CONFIG_PACKAGE_libopenssl=m
CONFIG_PACKAGE_libopenssl-conf=m
CONFIG_PACKAGE_liboping=m
CONFIG_PACKAGE_libpcap=m
CONFIG_PACKAGE_libpci=y
CONFIG_PACKAGE_libpcre=m
CONFIG_PACKAGE_libqrencode=m
CONFIG_PACKAGE_librrd1=m
CONFIG_PACKAGE_libsensors=y
CONFIG_PACKAGE_libstdcpp=y
CONFIG_PACKAGE_libsysfs=y
CONFIG_PACKAGE_libubus-lua=m
CONFIG_PACKAGE_libuci-lua=m
CONFIG_PACKAGE_libudev-zero=y
CONFIG_PACKAGE_libusb-1.0=y
CONFIG_PACKAGE_libustream-openssl=m
CONFIG_PACKAGE_libustream-wolfssl=m
CONFIG_PACKAGE_libwolfssl=m
CONFIG_PACKAGE_libxtables=m
CONFIG_PACKAGE_lm-sensors=y
CONFIG_PACKAGE_lsblk=y
CONFIG_PACKAGE_lua=m
CONFIG_PACKAGE_luci=m
CONFIG_PACKAGE_luci-app-adblock=m
CONFIG_PACKAGE_luci-app-ddns=m
CONFIG_PACKAGE_luci-app-firewall=m
CONFIG_PACKAGE_luci-app-nlbwmon=m
CONFIG_PACKAGE_luci-app-opkg=m
CONFIG_PACKAGE_luci-app-p910nd=m
CONFIG_PACKAGE_luci-app-sqm=m
CONFIG_PACKAGE_luci-app-statistics=m
CONFIG_PACKAGE_luci-app-uhttpd=m
CONFIG_PACKAGE_luci-app-wireguard=m
CONFIG_PACKAGE_luci-app-wol=m
CONFIG_PACKAGE_luci-base=m
CONFIG_PACKAGE_luci-compat=m
CONFIG_PACKAGE_luci-i18n-adblock-de=m
CONFIG_PACKAGE_luci-i18n-adblock-en=m
CONFIG_PACKAGE_luci-i18n-base-de=m
CONFIG_PACKAGE_luci-i18n-base-en=m
CONFIG_PACKAGE_luci-i18n-ddns-de=m
CONFIG_PACKAGE_luci-i18n-ddns-en=m
CONFIG_PACKAGE_luci-i18n-firewall-de=m
CONFIG_PACKAGE_luci-i18n-firewall-en=m
CONFIG_PACKAGE_luci-i18n-nlbwmon-de=m
CONFIG_PACKAGE_luci-i18n-nlbwmon-en=m
CONFIG_PACKAGE_luci-i18n-opkg-de=m
CONFIG_PACKAGE_luci-i18n-opkg-en=m
CONFIG_PACKAGE_luci-i18n-p910nd-de=m
CONFIG_PACKAGE_luci-i18n-p910nd-en=m
CONFIG_PACKAGE_luci-i18n-sqm-de=m
CONFIG_PACKAGE_luci-i18n-sqm-en=m
CONFIG_PACKAGE_luci-i18n-statistics-de=m
CONFIG_PACKAGE_luci-i18n-statistics-en=m
CONFIG_PACKAGE_luci-i18n-uhttpd-de=m
CONFIG_PACKAGE_luci-i18n-uhttpd-en=m
CONFIG_PACKAGE_luci-i18n-wireguard-de=m
CONFIG_PACKAGE_luci-i18n-wireguard-en=m
CONFIG_PACKAGE_luci-i18n-wol-de=m
CONFIG_PACKAGE_luci-i18n-wol-en=m
CONFIG_PACKAGE_luci-lib-base=m
CONFIG_PACKAGE_luci-lib-ip=m
CONFIG_PACKAGE_luci-lib-ipkg=m
CONFIG_PACKAGE_luci-lib-jsonc=m
CONFIG_PACKAGE_luci-lib-nixio=m
CONFIG_PACKAGE_luci-mod-admin-full=m
CONFIG_PACKAGE_luci-mod-network=m
CONFIG_PACKAGE_luci-mod-status=m
CONFIG_PACKAGE_luci-mod-system=m
CONFIG_PACKAGE_luci-proto-ipv6=m
CONFIG_PACKAGE_luci-proto-ppp=m
CONFIG_PACKAGE_luci-proto-wireguard=m
CONFIG_PACKAGE_luci-ssl-openssl=m
CONFIG_PACKAGE_luci-theme-bootstrap=m
CONFIG_PACKAGE_luci-theme-openwrt-2020=m
CONFIG_PACKAGE_netifd-netinfo=m
CONFIG_PACKAGE_netperf=y
CONFIG_PACKAGE_nlbwmon=m
CONFIG_PACKAGE_ntfs-3g=m
CONFIG_PACKAGE_ntfs-3g-utils=m
CONFIG_PACKAGE_openssl-util=m
# CONFIG_PACKAGE_openwrt-keyring is not set
CONFIG_PACKAGE_opkg=m
CONFIG_PACKAGE_p910nd=m
CONFIG_PACKAGE_pciids=y
CONFIG_PACKAGE_pciutils=y
CONFIG_PACKAGE_qrencode=m
CONFIG_PACKAGE_resize2fs=y
CONFIG_PACKAGE_rpcd=m
CONFIG_PACKAGE_rpcd-mod-file=m
CONFIG_PACKAGE_rpcd-mod-iwinfo=m
CONFIG_PACKAGE_rpcd-mod-luci=m
CONFIG_PACKAGE_rpcd-mod-rrdns=m
CONFIG_PACKAGE_rrdtool1=m
CONFIG_PACKAGE_rt2800-pci-firmware=y
CONFIG_PACKAGE_rt2800-usb-firmware=y
CONFIG_PACKAGE_slh-misc-16m=m
CONFIG_PACKAGE_slh-misc-32m=m
CONFIG_PACKAGE_slh-misc-8m=m
CONFIG_PACKAGE_slh-misc-luci-statistics=m
CONFIG_PACKAGE_slh-misc-qos=m
CONFIG_PACKAGE_speedtest-netperf=y
CONFIG_PACKAGE_sqm-scripts=m
CONFIG_PACKAGE_sqm-scripts-extra=m
CONFIG_PACKAGE_sysfsutils=y
CONFIG_PACKAGE_tc-tiny=m
CONFIG_PACKAGE_tcpdump-mini=m
CONFIG_PACKAGE_terminfo=y
CONFIG_PACKAGE_tftp-hpa=m
CONFIG_PACKAGE_tftpd-hpa=m
CONFIG_PACKAGE_uboot-envtools=m
CONFIG_PACKAGE_uhttpd=m
CONFIG_PACKAGE_uhttpd-mod-ubus=m
CONFIG_PACKAGE_umdns=m
CONFIG_PACKAGE_usbutils=y
# CONFIG_PACKAGE_usign is not set
CONFIG_PACKAGE_wireguard-tools=m
CONFIG_PACKAGE_wireless-regdb=y
CONFIG_PACKAGE_wpa-cli=m
CONFIG_PACKAGE_wpad=m
CONFIG_PACKAGE_wpad-basic=m
CONFIG_PACKAGE_wpad-basic-wolfssl=m
CONFIG_PACKAGE_wpad-mini=m
CONFIG_PACKAGE_wpad-openssl=m
CONFIG_PACKAGE_wpad-wolfssl=m
CONFIG_PACKAGE_xtables-nft=m
CONFIG_PACKAGE_zlib=y
CONFIG_PREINITOPT=y
# CONFIG_SIGNATURE_CHECK is not set
# CONFIG_SIGNED_PACKAGES is not set
CONFIG_TARGET_KERNEL_PARTSIZE=32
CONFIG_TARGET_PREINIT_TIMEOUT=5
CONFIG_TARGET_ROOTFS_PARTSIZE=920
CONFIG_WPA_MSG_MIN_PRIORITY=3
CONFIG_WPA_RFKILL_SUPPORT=y
CONFIG_WPA_WOLFSSL=y
CONFIG_ZLIB_OPTIMIZE_SPEED=y
CONFIG_OPENSSL_NO_DEPRECATED=y
# CONFIG_OPENSSL_WITH_DEPRECATED is not set
```

Reply:
we need to understand the problem with x86 but i think we are on the right track

Reply:
x86_generic builds fine (with wireguard enabled).

I did not test x86_legacy and x86_geode.

Reply:
Does a x64 system build also fine?

Reply:
No, x86_64 does not build - tested diffconfig in the post below. Error message in https://github.com/openwrt/openwrt/pull/9526#issuecomment-1075813365